controllers/base_controller.go

```diff
@@ -98,47 +98,103 @@ func (r *BaseReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
 		return ctrl.Result{}, err
 	}
 
-	spiffeId := r.makeSpiffeId(obj)
-	parentId := r.makeParentId(obj)
-	if spiffeId == "" || parentId == "" {
+	myEntry := r.makeEntryForObject(obj)
+
+	if myEntry == nil {
 		// Object does not need an entry. This might be a change, so we need to delete any hanging entries.
+		reqLogger.V(1).Info("Deleting entries for pod that no longer needs an ID", "count", len(matchedEntries))
 		err := r.deleteAllEntries(ctx, reqLogger, matchedEntries)
 		return ctrl.Result{}, err
 	}
 
-	createEntryIfNotExistsResponse, err := r.SpireClient.CreateEntryIfNotExists(ctx, &common.RegistrationEntry{
-		Selectors: r.getSelectors(req.NamespacedName),
-		ParentId:  r.makeParentId(obj),
-		SpiffeId:  spiffeId,
-	})
+	var myEntryId string
 
-	if err != nil {
-		reqLogger.Error(err, "Failed to create or update spire entry")
-		return ctrl.Result{}, err
-	}
-	if !createEntryIfNotExistsResponse.Preexisting {
-		reqLogger.Info("Created new spire entry", "entry", createEntryIfNotExistsResponse.Entry)
+	if len(matchedEntries) == 0 {
+		createEntryIfNotExistsResponse, err := r.SpireClient.CreateEntryIfNotExists(ctx, myEntry)
+		if err != nil {
+			reqLogger.Error(err, "Failed to create or update spire entry")
+			return ctrl.Result{}, err
+		}
+		if !createEntryIfNotExistsResponse.Preexisting {
```

Comment:
Could remove negation and swap if/else blocks. Having a `Not` in the var name is heavy to parse enough already, without the extra negation.